internal/cmd/skupper/link/nonkube/link_status.go

```diff
@@ -2,28 +2,129 @@ package nonkube
+	}
+
+	cmd.linkHandler = fs.NewLinkHandler(cmd.namespace)
+	cmd.siteHandler = fs.NewSiteHandler(cmd.namespace)
+}
+
+func (cmd *CmdLinkStatus) ValidateInput(args []string) []error {
+	var validationErrors []error
+	opts := fs.GetOptions{LogWarning: false}
+	resourceStringValidator := validator.NewResourceStringValidator()
+
+	// Validate arguments name if specified
+	if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if len(args) == 1 {
+		if args[0] == "" {
+			validationErrors = append(validationErrors, fmt.Errorf("link name must not be empty"))
+		} else {
+			ok, err := resourceStringValidator.Evaluate(args[0])
+			if !ok {
+				validationErrors = append(validationErrors, fmt.Errorf("link name is not valid: %s", err))
+			} else {
+				cmd.linkName = args[0]
+			}
+		}
+	}
+	// Validate that there is a link with this name in the namespace
+	if cmd.linkName != "" {
+		link, err := cmd.linkHandler.Get(cmd.linkName, opts)
+		if link == nil || err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("link %s does not exist in namespace %s", cmd.linkName, cmd.namespace))
+		}
+	}
+	return validationErrors
 }
 
-func (cmd *CmdLinkStatus) ValidateInput(args []string) []error { return nil }
-func (cmd *CmdLinkStatus) InputToOptions()                     {}
 func (cmd *CmdLinkStatus) Run() error {
-	return fmt.Errorf("command not supported by the selected platform")
+	opts := fs.GetOptions{RuntimeFirst: true, LogWarning: true}
+	if cmd.linkName == "" {
+		links, err := cmd.linkHandler.List()
+		if links == nil || err != nil {
+			fmt.Println("no links found:")
+			return err
+		}
+
+		tw := tabwriter.NewWriter(os.Stdout, 8, 8, 1, '\t', tabwriter.TabIndent)
+		_, _ = fmt.Fprintln(tw, fmt.Sprintf("%s\t%s",
+			"NAME", "STATUS"))
+		for _, link := range links {
+			status := "Not Ready"
+			if link.IsConfigured() {
+				status = "Ok"
+			}
+			name := link.Name
+			for index, endpoint := range link.Spec.Endpoints {
+				if endpoint.Name == types.InterRouterRole {
+					name = name + "-" + link.Spec.Endpoints[index].Host
+				}
+			}
+			fmt.Fprintln(tw, fmt.Sprintf("%s\t%s",
+				name, status))
+		}
+		_ = tw.Flush()
+	} else {
+		link, err := cmd.linkHandler.Get(cmd.linkName, opts)
+		if link == nil || err != nil {
+			fmt.Println("No link found:")
+			return err
+		}
+		status := "Not Ready"
+		if link.IsConfigured() {
+			status = "Ok"
+		}
+
+		// get the site and determine role of router
+		endpointName := ""
+		sites, err := cmd.siteHandler.List()
+		if sites != nil && err == nil {
+			endPointType := types.InterRouterRole
+			if sites[0].Spec.Edge {
+				endPointType = types.EdgeRole
+			}
+
+			for index, endpoint := range link.Spec.Endpoints {
+				if endpoint.Name == endPointType {
+					endpointName = link.Spec.Endpoints[index].Host + ":" + link.Spec.Endpoints[index].Port
```

Comment:
I am still unable to see the endpoints.

Reply:
Do you have a site configured?   I look up the site to determine if it is an edge or inter-router to display the correct endpoints.    If there is no site no endpoints are displayed.    There is also a bug https://github.com/skupperproject/skupper/pull/1880  that causes a problem and needs to be fixed.    This is just incorrect name of the site directory.

Reply:
defaulting to use inter-router

Reply:
It works for me now. Only issue I am seeing is that the following message is displayed: `Site not initialized yet`.

Reply:
fixed